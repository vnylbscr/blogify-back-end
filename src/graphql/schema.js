import { makeExecutableSchema } from '@graphql-tools/schema';
import FileTypeDefs from './file/typeDefs.js';
import { UserResolvers, UserTypeDefs } from './user/index.js';
import { PostResolvers, PostTypeDefs } from './post/index.js';
import { CommentResolvers, CommentTypeDefs } from './comment/index.js';

const typeDefs = [UserTypeDefs, PostTypeDefs, CommentTypeDefs, FileTypeDefs];
const resolvers = [UserResolvers, PostResolvers, CommentResolvers];

const schema = makeExecutableSchema({
   typeDefs,
   resolvers,
});

export default schema;
