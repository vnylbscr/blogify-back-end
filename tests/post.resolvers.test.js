import { ApolloServer } from 'apollo-server-express';
import mongoose from 'mongoose';
import cloudinary from 'cloudinary';
import schema from '../graphql/schema.js';
import postResolvers from '../graphql/post/resolver.js';
import { GET_ALL_POSTS, GET_SINGLE_POST } from './queries.js';
import Auth from '../middleware/auth.js';

describe('POST resolvers', () => {
   const server = new ApolloServer({
      schema,
      context: ({ req }) => {
         const isAuth = Auth(req);
         return {
            isAuth,
         };
      },
   });
   beforeAll(async () => {
      mongoose
         .connect(process.env.DB_URL, {
            useUnifiedTopology: true,
            useNewUrlParser: true,
         })
         .then(() => console.log('MongoDB Connected.'))
         .catch((error) => console.log(error));
      mongoose.set('useCreateIndex', true);

      cloudinary.v2.config({
         cloud_name: process.env.CLOUD_NAME,
         api_key: process.env.CLOUD_API_KEY,
         api_secret: process.env.CLOUD_API_SECRET_KEY,
      });
   });

   test('should get all post from database', async () => {
      const result = await server.executeOperation({
         query: GET_SINGLE_POST,
         variables: {
            _id: '1',
         },
      });
      expect(result).toEqual('post not found');
   });
});
