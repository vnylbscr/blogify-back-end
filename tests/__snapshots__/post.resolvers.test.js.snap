// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST resolvers should get all post from database 1`] = `undefined`;
